transmart-core-db/src/main/groovy/org/transmartproject/db/tree/TreeNode.groovy

```diff
@@ -268,4 +276,26 @@ class TreeNode {
         }
     }
 
+    final String getDimension() {
+        switch (tableName) {
+            case 'concept_dimension':
+                return DimensionImpl.CONCEPT.name
+            case 'patient_dimension':
+                return DimensionImpl.PATIENT.name
+            case 'modifier_dimension':
+                def dimension = DimensionDescription.createCriteria().list {
```

Comment:
I'd hope GORM is able to cache such queries (in which case this use would be fine), but I'm not sure if it can or will in this case.